src/resolve/resolvectl.c

```diff
@@ -2531,11 +2504,14 @@ static int compat_parse_argv(int argc, char *argv[]) {
 
                 case ARG_TLSA:
                         arg_mode = MODE_RESOLVE_TLSA;
-                        if (service_family_from_string(arg_service_family) < 0) {
-                                log_error("Unknown service family \"%s\".", optarg);
-                                return -EINVAL;
+                        if (optarg) {
+                                if (service_family_is_valid(optarg))
+                                        arg_service_family = optarg;
+                                else {
+                                        log_error("Unknown service family \"%s\".", optarg);
+                                        return -EINVAL;
```

Comment:
This changes behaviour when --tlsa without the optional argument is specified. Before it'd reset `arg_service_family`, but now nothing happens.

Reply:
Indeed. Thanks. Updated.